### PR TITLE
Add Acknowledgement for QR Code Library & Table of Contents

### LINKED ISSUE #106: Docs: Update acknowledgements for QR code libraries
With reference to #105, we need to acknowledge the use of external libraries for the QR code feature.

### PR DESCRIPTION
**Pull request by dillontkh**
Resolves #106

### CODE REVIEW

**Review by ruijietay — APPROVED** [review]
LGTM 🤠

**Review by RyanNgWH — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -4,14 +4,15 @@ title: Developer Guide
...
 
 --------------------------------------------------------------------------------------------------------------------
 
 ## **Acknowledgements**
 
-* {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
-  original source as well}
+| Module                                  | Description                               |
+|-----------------------------------------|-------------------------------------------|
+| [ZXing](https://github.com/zxing/zxing) | Barcode image processing library for Java |
```
Comment (RyanNgWH) [inline docs/DeveloperGuide.md:15]: Should we include JavaFX, Jackson & JUnit5 here too? Similar to the one in the index file.

```suggestion
| [JavaFX](https://openjfx.io/)                   | UI generation for Java |
| [Jackson](https://github.com/FasterXML/jackson) | Json processing library for Java |
| [JUnit5](https://github.com/junit-team/junit5)  | Automated testing library for Java |
| [ZXing](https://github.com/zxing/zxing)         | Barcode image processing library for Java |
```

**Review by RyanNgWH — APPROVED** [review]
Just a thought on including the other references in the DG as well. Else, LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (dillontkh) [reply docs/DeveloperGuide.md]: Good idea

**Pull request merged**